File: Cpp/Source/Field.Test/source/ImagineThisIsAClientInADifferentRepositoryTest.cpp

#include "Field/include/ImagineThisIsAClientInADifferentRepository.h"

#include <gtest/gtest.h>

TEST( singleItem_numberOfProductsInTheCart, ImagineThisIsAClientInADifferentRepositoryTest )
{
    ImagineThisIsAClientInADifferentRepository client;

    EXPECT_EQ(std::string("Total price is 50 euro"), client.formattedTotalPrice(50));
}

File: Java/src/test/java/com/pduda/field/ImagineThisIsAClientInADifferentRepositoryTest.java
package com.pduda.field;

import org.junit.Assert;
import org.junit.Test;

public class ImagineThisIsAClientInADifferentRepositoryTest {


    @Test
    public void singleItem_numberOfProductsInTheCart() {
        ImagineThisIsAClientInADifferentRepository client = new ImagineThisIsAClientInADifferentRepository();

        Assert.assertEquals("Total price is 50 euro", client.formattedTotalPrice(50));
    }

}
